cmds/core/shasum/shasum.go

```diff
@@ -9,79 +9,76 @@ import (
 	"crypto/sha256"
 	"fmt"
 	"io"
+	"log"
 	"os"
 
 	"github.com/spf13/pflag"
 )
 
+var (
+	algorithm = pflag.IntP("algorithm", "a", 1, "SHA algorithm, valid args are 1 and 256")
+	help      = pflag.BoolP("help", "h", false, "Show this help and exit")
+)
+
+var usage = "Usage:\nshasum -a <algorithm> <File Name>"
+
 func helpPrinter() {
-	fmt.Printf("Usage:\nshasum -a <algorithm> <File Name>\n")
+	fmt.Println(usage)
 	pflag.PrintDefaults()
-	os.Exit(0)
 }
 
-func versionPrinter() {
-	fmt.Println("shasum utility, URoot Version.")
-	os.Exit(0)
-}
-
-func getInput(fileName string) (input []byte, err error) {
+func getInput(r io.Reader, fileName string) (input []byte, err error) {
 	if fileName != "" {
 		return os.ReadFile(fileName)
 	}
-	return io.ReadAll(os.Stdin)
+	return io.ReadAll(r)
```

Comment:
same thing. Let's get rid of the io.ReadAll as we did in md5sum. 

In fact, thinking about it, could we have a program, called sum, that does this
sum [-type someName] etc. etc. 

And -type could be md5 (default), sha256, etc.

I'd suggest redoing this program along the same lines we redid md5sum, if you don't wish to do the merge.

Reply:
I tried to redoing this program like the redid in md5sum. Did you noticed?